lib/services.identity.js

```diff
@@ -596,7 +613,19 @@ function resolveIdentitySlug(name, options, callback) {
 
 function updateIdentity(req, operation, callback) {
   async.auto({
-    getId: getIdentityIdFromUrl.bind(null, req),
+    getId: function(callback) {
+      if(req.params.identity && operation.id) {
+        return callback(new BedrockError(
+          'The identity could not be updated.', 'UpdateIdentityFailed', {
+          httpStatusCode: 400,
+          public: true
+        }));
+      }
```

Comment:
Eh, it's unclear what the problem is. It should probably be a validation error. At the very least we should say `'The identity could not be updated due to invalid input; the `id` property must not be provided.'`.

The real fix would probably be something like running a real quick `validate.validateInstance(<on-the-fly schema here>, ...)` check that just makes sure `id` is not in the operation and then return the error from that if there is one.